Mage/src/main/java/mage/abilities/effects/keyword/ExploreSourceEffect.java

```diff
@@ -49,6 +59,7 @@ public static String getRuleText(boolean showAbilityHint, String whosExplores) {
 
     private String sourceName = "it";
     private boolean showAbilityHint = true;
+    private byte numTimesExplored = 0;
```

Comment:
1. You are added text about outside logic here. Ability contains info about times explored, but real effect logic still lay down in the parent ability as two added effects). That's wrong. If you add double times logic in effect then it must be here too (e.g. call `explorePermanent` multiple times in apply method).

2. Also it have non standard type: `byte`. Try to use standard types like `int` (if you try to find `byte` in xmage's code, can't find real usage of it).

Reply:
Understandable! Thank you for the help!